src/Workspaces/Core/Portable/Shared/Extensions/SyntaxTreeExtensions.cs

```diff
@@ -72,6 +72,46 @@ public static bool IsScript(this SyntaxTree syntaxTree)
             return default;
         }
 
+        /// <summary>
+        /// Returns the first Token that passes <paramref name="predicate"/> within supplied <paramref name="span"/>.
+        /// </summary>
+        public static async Task<SyntaxToken> GetLeftmostTokenInSpanAsync(
+            this SyntaxTree syntaxTree,
+            TextSpan span, 
+            Predicate<SyntaxToken> predicate,
+            CancellationToken cancellationToken,
+            bool findInsideTrivia = false,
+            bool includeSkipped = true,
+            bool includeDirectives = false,
+            bool includeDocumentationComments = false)
+        {
+            Contract.ThrowIfNull(syntaxTree);
+
+            if (span.Start >= syntaxTree.Length)
+            {
+                return default;
+            }
+
+            var root = await syntaxTree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+            var token = root.FindToken(span.Start, findInsideTrivia);
+
+            while (!predicate(token))
+            {
+                if (token.FullSpan.End >= syntaxTree.Length)
```

Comment:
shouldn't this also check the given span to see token is outside of the given span?

Reply:
Yep, that's quite an oversight. Will fix.